### DISCUSSION TITLE
sBTC Design - API / SDK

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-archive/sbtc-developer-release#320] [Ideas]
This is a discussion post meant to be edited with research pertaining to sBTC API and SDK designs. This will be informed by the content of the hackathon that took place this last weekend. This won't be part of the sBTC SIP but will be critical when we get to the design phase of our outwardly facing software next week.

**Comment by AshtonStephens** [comment]
Currently under investigation by @radicleart

**Comment by mijoco-btc** [comment]
Meeting with @janniks , @radicleart and @kenrogers,

1. sBTC Javascript libs.
The code currently in the sbtc branch of stacks.js will become part of @stacks/transactions or a new package @stacks/sbtc. Janniks is looking into simplifying the process of UTXO selection for building the OP_RETURN transactions by removing the change address and deferring functionality to the wallet to add the fee. Also looking into exporting network objects (static exports) more akin to the btc-scure library.

Standard features like providing methods for building the payloads and giving developer control over the process is already part of the DR release code.

2. New requirements - from SP review process
Changes from the SIP will be tracked and included in the evolving SDK. E.g. ability to handle commit reveal will imply the need for tools to submit commit transactions and tap script data to a clarity contract or to look up previously submitted commitment transactions.

3. APIs
Discussed the possibility of accessing a public electrumx or equivalent server to facilitate sBTC operations. 
It may make sense to expose some specific endpoints via the bridge which proxy to an electrumx like server.
 
4. Documentation
In process of being consolidated and moved to GitBook.

References
1. [add sBTC functionality in stacks.js](https://github.com/hirosystems/stacks.js/pull/1554)
2. [add-sbtc-contracts](https://github.com/hirosystems/stacks.js/blob/feat/add-sbtc-contracts/packages/sbtc/src/api.ts)
3. [sbtc 0.1.x developer release](https://www.npmjs.com/package/sbtc)
4. [sbtc-bridge-lib](https://github.com/stacks-network/sbtc-bridge-api/tree/main/sbtc-bridge-lib)

**Comment by AshtonStephens** [comment]
I'm moving this discussion from "research" to "design" because it is not part of the sBTC SIP, and is more related to implementation details.
